File: java-ee-servlet/src/main/java/linkboard/service/LinkGroupService.java
package linkboard.service;

import linkboard.data.dao.LinkGroupDao;
import linkboard.data.entity.LinkGroupEntity;
import linkboard.data.entity.UserAccountEntity;

import java.util.List;

public class LinkGroupService
{
    // TODO: Use CDI
    private static final LinkGroupDao linkGroupDao = new LinkGroupDao();

    public List<LinkGroupEntity> getAllForUser(UserAccountEntity user)
    {
        return linkGroupDao.findAllForUser(user);
    }

    public LinkGroupEntity saveLinkGroup(LinkGroupEntity linkGroup)
    {
        return linkGroupDao.save(linkGroup);
    }
}

File: go-vanilla/src/go/service/LinkGroupService.go
package service

import (
    "go/data/dao"
    "go/data/model"
)

func GetLinkGroupsForUser(user *model.User) *[]model.LinkGroup {
    return dao.FindLinkGroupsForUser(user)
}

func SaveLinkGroup(group *model.LinkGroup) *model.LinkGroup {
    return dao.SaveLinkGroup(group)
}